consensus/src/liveness/proposal_generator.rs

```diff
@@ -536,25 +548,17 @@ impl ProposalGenerator {
 
         let mut execution_backpressure_applied = false;
         if let Some(config) = &self.pipeline_backpressure_config.execution {
-            if pipeline_pending_latency.as_millis()
-                > config.back_pressure_pipeline_latency_limit_ms as u128
-            {
-                let execution_backpressure = self
-                    .pipeline_backpressure_config
-                    .get_execution_block_size_backoff(
-                        &self
-                            .block_store
-                            .get_recent_block_execution_times(config.num_blocks_to_look_at),
-                    );
-                if let Some(execution_backpressure_block_size) = execution_backpressure {
-                    values_max_block_txns_after_filtering.push(
-                        (execution_backpressure_block_size as f64
-                            * config.reordering_ovarpacking_factor.max(1.0))
-                            as u64,
-                    );
-                    values_max_txns_from_block_to_execute.push(execution_backpressure_block_size);
-                    execution_backpressure_applied = true;
-                }
+            let execution_backpressure = self
+                .pipeline_backpressure_config
+                .get_execution_block_size_backoff(
+                    &self
+                        .block_store
+                        .get_recent_block_execution_times(config.num_blocks_to_look_at),
+                    self.max_block_txns_after_filtering,
```

Comment:
nit: 
`values_max_block_txns_after_filtering` contains `self.max_block_txns_after_filtering` and `execution_backpressure`. In the end we calculate `values_max_block_txns_after_filtering`.min().  Do we have to explicitly pass `self.max_block_txns_after_filtering` again as input here to perform the min operation inside `get_execution_block_size_backoff`?

Reply:
we are passing this purely to affect visibility - i.e. how counters are logged etc. 

seems reasonable to treat each backpressure as kicking off independently, and then the most aggressive one is the one actually used - so that's why the logic is as is.